### PR TITLE
Handle ldap race condition gracefully

### PR DESCRIPTION
**Pull request by maorfr**

### CODE REVIEW

File: utils/gitlab_api.py [hunk 1]
```diff
@@ -70,5 +73,16 @@ def create_delete_user_mr(self, username, path):
...
         self.create_branch(branch_name, target_branch)
-        self.delete_file(branch_name, path, title)
+
+        try:
+            self.delete_file(branch_name, path, title)
+        except gitlab.exceptions.GitlabCreateError as e:
+            self.delete_branch(branch_name)
+            if str(e) != "400: A file with this name doesn't exist":
+                raise e
+            print(
```
Comment (jmelis) [inline utils/gitlab_api.py]: use `logging.info`

File: utils/gitlab_api.py [hunk 2]
```diff
@@ -70,5 +73,16 @@ def create_delete_user_mr(self, username, path):
             return
 
         self.create_branch(branch_name, target_branch)
-        self.delete_file(branch_name, path, title)
+
+        try:
+            self.delete_file(branch_name, path, title)
+        except gitlab.exceptions.GitlabCreateError as e:
+            self.delete_branch(branch_name)
```
Comment (jmelis) [inline utils/gitlab_api.py]: This is very clever! I would've forgotten to do this...

File: utils/gitlab_api.py (hunk 1 above)
Comment (maorfr) [reply utils/gitlab_api.py]: Doh!

**Comment by maorfr** [comment]
https://issues.redhat.com/browse/APPSRE-408

**Pull request merged**